src/librustc_errors/emitter.rs

```diff
@@ -347,8 +347,7 @@ impl EmitterWriter {
 
         // Sort the annotations by (start, end col)
         let mut annotations = line.annotations.clone();
-        annotations.sort();
-        annotations.reverse();
+        annotations.sort_by(|a,b| b.start_col.cmp(&a.start_col));
```

Comment:
If you have a list of annotations (A1, A2, C1, C2, B1, B2) where the letter signifies the span, since we are only sorting by the span, the order of the elements with the same span will never change. Since we are reversing the ordering (`|a, b|` but `b.cmp(a)`), you get (C1, C2, B1, B2, A1, A2). All the elements with the same span are still ordered first to last, but all the elements with different spans are ordered by their spans in last to first order. Last to first order is important, because the jiggly lines and `|` are on the left, so the rightmost span needs to be rendered first, otherwise the lines would end up needing to go over a message.

If the above is comprehensible, I think it can be copy pasted as the requested comment ;)